llvm/lib/CodeGen/MIRParser/MIRParser.cpp

```diff
@@ -398,20 +400,25 @@ void MIRParserImpl::computeFunctionProperties(MachineFunction &MF) {
       }
     }
   }
-  if (!HasPHI)
+
+  // Don't overwrite NoPHIs if the input MIR explicitly set it to false
+  if (YamlMF.NoPHIs && !HasPHI)
     Properties.set(MachineFunctionProperties::Property::NoPHIs);
+
   MF.setHasInlineAsm(HasInlineAsm);
 
   if (HasTiedOps && AllTiedOpsRewritten)
     Properties.set(MachineFunctionProperties::Property::TiedOpsRewritten);
 
-  if (isSSA(MF))
+  // Don't overwrite IsSSA if the input MIR explicitly set it to false
+  if (YamlMF.IsSSA && isSSA(MF))
```

Comment:
Could we report a warning what the flag and the computed value don't agree?

Reply:
I think this needs to be a hard error, MIR should not be permissive

Reply:
This will just produce noise we don't really want printing during tests

Reply:
Fair enough

Reply:
Changed it to produce an error if the explicit property conflicts with the input MIR